lib/src/stream_extensions.dart

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+/// Utility extensions on [Stream].
+extension StreamExtensions<T> on Stream<T> {
+  Stream<List<T>> slices(int length) {
+    if (length < 1) throw RangeError.range(length, 1, null, 'length');
+
+    var slice = <T>[];
```

Comment:
I'm actually more concerned that `async*` methods are delayed in starting, so you might miss events from broadcast streams. So two reasons not to do it.

I tried writing it without a transformer, and it's definitely not shorter. Probably marginally more efficient since it doesn't need to allocate the sink object, but not something I expect anyone to notice.
So, OK.